contracts/tonomy/src/tonomy.cpp

```diff
@@ -497,4 +497,63 @@ namespace tonomysystem
           .send();
    }
 
+   void tonomy::staketokens(name account_name, asset quantity) {
+      require_auth(account_name);
+      eosio::check(quantity.amount > 0, "Stake quantity must be greater than zero.");
```

Comment:
need to do additional checks on the asset like we do here: https://github.com/Tonomy-Foundation/Tonomy-Contracts/blob/master/contracts/vesting.tmy/src/vesting.tmy.cpp#L5